src/cameras/src/pm/Acquisition.cpp

```diff
@@ -238,7 +241,13 @@ void pm::Acquisition<F, C>::frameWriterThread() {
                             }
                             break;
                     }
-                    file->WriteFrame<F>(frame);
+
+
+                    writerPool.AddTask([this, &file, frame] {
```

Comment:
if you're wondering about the syntax
```cpp
[this, &file, frame] {
   ...
}
```
It's the capture settings for the lambda that says to capture `this` by value, since it's a pointer, `&file` by reference and `frame` by value.

might be able to use
```cpp
[=] {
  ...
}
```
instead but this makes it clear that `frame` needs to be by value, it's a pointer anyway but `frame` gets nulled out after the lambda is created so I did this to maybe be safe, idk.